pages/upgrading_an_app.md

```diff
@@ -25,23 +26,41 @@ To upgrade your application, type:
 
 `yo jhipster:upgrade`
 
-If you want to reset your codebase using the current installed JHipster version, type:
+If you want to run the upgrade sub-generator even if no new JHipster version is available, type:
 
 `yo jhipster:upgrade --force`
 
+## Requirements
+
 For this sub-generator to work you need to have `git` installed from [http://git-scm.com](http://git-scm.com/).
 
+## Step-by-step explanation
+
 Below are the steps processed by the JHipster upgrade sub-generator:
 
 1. Check if there is a new version of JHipster available (not applicable if you are using `--force`).
-2. Check If the application is already initialized as a `git` repository, or else JHipster will initialize one for you and commit the current codebase to the master branch.
+2. Check if the application is already initialized as a `git` repository, or else JHipster will initialize one for you and commit the current codebase to the master branch.
 3. Check to ensure that there are no un-committed local changes in the repository. The process will exit if there are un-committed changes found.
-4. Create and checkout a new branch called `jhipster_upgrade`.
-5. Upgrade JHipster to the latest available version globally.
-6. Clean the current project directory.
-7. Re-generate the application using the `yo jhipster --force --with-entities` command.
-8. Commit the generated code to the `jhipster_upgrade` branch.
-9. Merge the `jhipster_upgrade` branch back to the original branch from where the `yo jhipster:upgrade` command was launched.
-10. Now you just need to proceed with resolving merge conflicts if there are any.
+4. Check if `jhipster_upgrade` branch exists ; if not, branch is created. Some details about this step is provided in "First upgrade" section.
+5. Checkout `jhipster_upgrade` branch.
+6. Upgrade JHipster to the latest available version globally.
+7. Clean the current project directory.
+8. Re-generate the application using the `yo jhipster --force --with-entities` command.
+9. Commit the generated code to the `jhipster_upgrade` branch.
+10. Merge the `jhipster_upgrade` branch back to the original branch from where the `yo jhipster:upgrade` command was launched.
+11. Now you just need to proceed with resolving merge conflicts if there are any.
 
 Congratulations, your application is now upgraded with the latest version of JHipster!
+
+![GitGraph]{{ site.url }}/images/upgrade_gitgraph.png)
+
+## Advice
```

Comment:
I finally have jekyll working (thx to Docker), and I have the same problem. I'm having a look